Mathlib/Analysis/SpecialFunctions/Artanh.lean

```diff
@@ -0,0 +1,164 @@
+/-
+Copyright (c) 2025 Yuval Filmus. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yuval Filmus
+-/
+module
+
+public import Mathlib.Analysis.SpecialFunctions.Log.Basic
+
+/-!
+# Inverse of the tanh function
+
+In this file we define an inverse of tanh as a function from ℝ to (-1, 1).
+
+## Main definitions
+
+- `Real.artanh`: An inverse function of `Real.tanh` as a function from ℝ to (-1, 1).
+
+- `Real.tanhPartialEquiv`: `Real.tanh` and `Real.artanh` bundled as a `PartialEquiv`
+  from ℝ to (-1, 1).
+
+## Main Results
+
+- `Real.tanh_artanh`, `Real.artanh_tanh`: tanh and artanh are inverse in the appropriate domains.
+
+- `Real.tanh_bijOn`, `Real.tanh_injective`, `Real.tanh_surjOn`: `Real.tanh` is
+  bijective, injective and surjective as a function from ℝ to (-1, 1)
+
+- `Real.artanh_bijOn`, `Real.artanh_injOn`, `Real.artanh_surjOn`: `Real.artanh` is bijective,
+  injective and surjective as a function from (-1, 1) to ℝ
+
+## Tags
+
+artanh, arctanh, argtanh, atanh
+-/
+
+@[expose] public section
+
+
+noncomputable section
+
+open Function Filter Set
+
+open scoped Topology
+
+namespace Real
+
+variable {x y : ℝ}
+
+/-- `artanh` is defined using a logarithm, `artanh x = log √((1 + x) / (1 - x))`. -/
+@[pp_nodot]
+def artanh (x : ℝ) :=
+  log √((1 + x) / (1 - x))
+
+theorem artanh_eq_half_log {x : ℝ} (hx : x ∈ Icc (-1) 1) :
+    artanh x = 1 / 2 * log ((1 + x) / (1 - x)) := by
+  rw [artanh, log_sqrt <| div_nonneg (by grind) (by grind), one_div_mul_eq_div]
+
+theorem exp_artanh {x : ℝ} (hx : x ∈ Ioo (-1) 1) : exp (artanh x) = √((1 + x) / (1 - x)) :=
+  exp_log <| sqrt_pos_of_pos <| div_pos (by grind) (by grind)
+
+@[simp]
+theorem artanh_zero : artanh 0 = 0 := by simp [artanh]
+
+theorem sinh_artanh {x : ℝ} (hx : x ∈ Ioo (-1) 1) : sinh (artanh x) = x / √(1 - x ^ 2) := by
+  have : 0 < √((1 + x) / (1 - x)) := sqrt_pos_of_pos <| div_pos (by grind) (by grind)
+  have := sq_sub_sq 1 x
+  grind [artanh, sinh_eq, exp_neg, exp_log, sqrt_div, sqrt_mul]
+
+theorem cosh_artanh {x : ℝ} (hx : x ∈ Ioo (-1) 1) : cosh (artanh x) = 1 / √(1 - x ^ 2) := by
+  have : 0 < √((1 + x) / (1 - x)) := sqrt_pos_of_pos <| div_pos (by grind) (by grind)
+  have := sq_sub_sq 1 x
+  grind [artanh, cosh_eq, exp_neg, exp_log, sqrt_div, sqrt_mul]
```

Comment:
This fixes the build error:
```suggestion
theorem sinh_artanh {x : ℝ} (hx : x ∈ Ioo (-1) 1) : sinh (artanh x) = x / √(1 - x ^ 2) := by
  have : 0 < √((1 + x) / (1 - x)) := sqrt_pos_of_pos <| div_pos (by grind) (by grind)
  rw [← one_pow, sq_sub_sq 1 x, sqrt_mul]
    <;> grind [artanh, sinh_eq, exp_neg, exp_log, sqrt_div]

theorem cosh_artanh {x : ℝ} (hx : x ∈ Ioo (-1) 1) : cosh (artanh x) = 1 / √(1 - x ^ 2) := by
  have : 0 < √((1 + x) / (1 - x)) := sqrt_pos_of_pos <| div_pos (by grind) (by grind)
  rw [← one_pow, sq_sub_sq 1 x, sqrt_mul]
    <;> grind [artanh, cosh_eq, exp_neg, exp_log, sqrt_div]
```
As I mentioned previously, unfortunately `grind` proofs can break when Mathlib updates to a newer Lean version :(